internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -0,0 +1,560 @@
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MongoCluster.MongoClustersClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var state MongoClusterResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := mongoclusters.NewMongoClusterID(subscriptionId, state.ResourceGroupName, state.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameter := mongoclusters.MongoCluster{
+				Location:   azure.NormalizeLocation(state.Location),
+				Properties: &mongoclusters.MongoClusterProperties{},
+			}
+
+			if state.AdministratorUserName != "" {
+				parameter.Properties.Administrator = &mongoclusters.AdministratorProperties{
+					UserName: pointer.To(state.AdministratorUserName),
+					Password: pointer.To(state.AdministratorPassword),
+				}
+			}
+
+			if state.CreateMode != "" {
+				parameter.Properties.CreateMode = pointer.To(mongoclusters.CreateMode(state.CreateMode))
+			}
+
+			if v, ok := metadata.ResourceData.GetOk("preview_features"); ok {
+				parameter.Properties.PreviewFeatures = expandPreviewFeatures(v.([]interface{}))
+			}
+
+			if _, ok := metadata.ResourceData.GetOk("shard_count"); ok {
+				parameter.Properties.Sharding = &mongoclusters.ShardingProperties{
+					ShardCount: pointer.To(state.ShardCount),
+				}
+			}
+
+			if state.CreateMode == string(mongoclusters.CreateModeGeoReplica) {
+				parameter.Properties.ReplicaParameters = &mongoclusters.MongoClusterReplicaParameters{
+					SourceLocation:   state.SourceLocation,
+					SourceResourceId: state.SourceServerId,
+				}
+			}
+
+			if state.ComputeTier != "" {
+				parameter.Properties.Compute = &mongoclusters.ComputeProperties{
+					Tier: pointer.To(state.ComputeTier),
+				}
+			}
+
+			if state.HighAvailabilityMode != "" {
+				parameter.Properties.HighAvailability = &mongoclusters.HighAvailabilityProperties{
+					TargetMode: pointer.To(mongoclusters.HighAvailabilityMode(state.HighAvailabilityMode)),
+				}
+			}
+
+			if state.PublicNetworkAccessEnabled {
+				parameter.Properties.PublicNetworkAccess = pointer.To(mongoclusters.PublicNetworkAccessEnabled)
+			} else {
+				parameter.Properties.PublicNetworkAccess = pointer.To(mongoclusters.PublicNetworkAccessDisabled)
+			}
+
+			if _, ok := metadata.ResourceData.GetOk("storage_size_in_gb"); ok {
+				parameter.Properties.Storage = &mongoclusters.StorageProperties{
+					SizeGb: pointer.To(state.StorageSizeInGb),
+				}
+			}
+
+			if _, ok := metadata.ResourceData.GetOk("tags"); ok {
```

Comment:
please use `Tags` instead of `metadata.ResourceData.GetOk("tags")`

Reply:
Fixed.